src/main/java/Storage.java

```diff
@@ -0,0 +1,151 @@
+
+    /**
+     * Adds file into the directory specified.
+     *
+     * @return whether file has been successfully added into the hard disk.
+     */
+    public boolean addFile() {
+        File directory = new File(dir);
+
+        try {
+            if (directory.mkdir()) {
+                storageFile.createNewFile();
+
+                return true;
+            }
+        } catch (IOException e) {
+            System.out.println("An error occurred while trying to create the file.");
+        }
+
+        return false;
+    }
+
+    /**
+     * Indicates if the saved file is existing.
+     *
+     * @return true if the saved file does exist, false otherwise.
+     */
+    public boolean checkFileExists() {
+        return storageFile.exists();
+    }
+
+    /**
+     * Returns the stored task list from the hard disk.
+     *
+     * @return an ArrayList containing the tasks stored in the hard disk.
+     * @throws FileNotFoundException If the storageFile is not located in its rightful directory.
+     */
+    public ArrayList<Task> retrieveTasks() throws FileNotFoundException {
+
+        ArrayList<Task> tasks = new ArrayList<>();
+
+        Scanner sc = new Scanner(storageFile);
+
+        while (sc.hasNextLine()) {
+            final String task = sc.nextLine(); //Each task is listed in a single line
+
+            try {
+                char type = task.charAt(0);
+                String[] descriptions = task.split("::"); //:: demarcates a different field
+
+                switch (type) {
+                case 'T':
+                    tasks.add(new ToDo(descriptions[2], descriptions[1].matches("1")));
+                    break;
+
+                case 'D':
+                    //Events.Deadline
+                    tasks.add(new Deadline(descriptions[2], descriptions[3], descriptions[1].matches("1")));
+                    break;
+
+                case 'E':
+                    //Events.Event
+                    tasks.add(new Event(descriptions[2], descriptions[3], descriptions[1].matches("1")));
+                    break;
+
+                default:
+                    System.out.println("Wrong file format");
+                }
+
+            } catch (StringIndexOutOfBoundsException e) {
+                //Formatting error in saved file
+                e.printStackTrace();
+                System.out.println("File corrupted.");
+            }
+        }
+
+        sc.close();
+
+        return tasks;
+    }
+
+    /**
+     * Updates the saved data in the disk.
+     *
+     * @param tasks is the current list of tasks the user has.
+     * @param isAddingTask indicates whether the user is adding a new task or overwriting the file.
+     * @return true if file is successfully updated.
+     * @throws IOException When writer is called upon to write after being closed.
+     */
+    public boolean updateData(ArrayList<Task> tasks, boolean isAddingTask) throws IOException {
```

Comment:
Perhaps you can give a more intuitive variable name for isAddingTask? (e.g: isOverwriting, isNewTask)